src/mono/mono/eventpipe/ds-rt-mono.c

```diff
@@ -236,31 +238,34 @@ ipc_transport_get_default_name (
 #endif // __APPLE__
 	{
 		// Get a temp file location
-		result = ep_rt_temp_path_get (format_buffer, name_len);
-		if (result == 0) {
+		format_result = ep_rt_temp_path_get (format_buffer, name_len);
+		if (format_result == 0) {
 			DS_LOG_ERROR_0 ("ep_rt_temp_path_get failed");
 			ep_raise_error ();
 		}
 
-		EP_ASSERT (result <= name_len);
+		EP_ASSERT (format_result <= name_len);
 	}
 
-	result = snprintf(name, name_len, "%s%s-%d-%llu-%s", format_buffer, prefix, id, (unsigned long long)disambiguation_key, suffix);
-	if (result <= 0 || (uint32_t)result > name_len) {
+	format_result = snprintf(name, name_len, "%s%s-%d-%llu-%s", format_buffer, prefix, id, (unsigned long long)disambiguation_key, suffix);
+	if (format_result <= 0 || (uint32_t)format_result > name_len) {
 		DS_LOG_ERROR_0 ("name buffer to small");
 		ep_raise_error ();
 	}
 
+	result = true;
+
 ep_on_exit:
 	free (format_buffer);
-	return;
+	return result;
 
 ep_on_error:
+	EP_ASSERT (result == false);
```

Comment:
nit
```suggestion
	EP_ASSERT (!result);
```

Reply:
This is a convention I have used through the library for asserts, being explicit (CoreClr does similar in their precondition checks against NULL etc), so if we change this, it should be done everywhere and not in this location. I believe the current way is clear but if we find it bad, then it will be better to do overhaul in separate PR.

Reply:
I guess as long as they're just comparing with `false` this is fine.  I am worried about `x == true` though since `bool` could have more than 2 possible values.

Reply:
I made a pass over library and simplified all checks against bool to eliminate any potential issues due to that.